base/server/python/pki/server/instance.py

```diff
@@ -511,11 +511,27 @@ def get_sslserver_cert_nickname(self):
         # If not available, load SSL server cert nickname from serverCertNick.conf
         # TODO: Remove serverCertNick.conf
 
-        server_cert_nick_conf = os.path.join(self.conf_dir, 'serverCertNick.conf')
+        logger.info('Getting serverCertNickFile from server.xml')
+
+        document = etree.parse(self.server_xml, parser)
+        server = document.getroot()
+        server_cert_nick_conf = None
+
+        connectors = server.findall('Service/Connector')
```

Comment:
Since you need only one you can use find and the add the attribute (should be something like  `server.find("Service/Connector[@secure='true']")`. This way you skip the for. 

For reference [lxml doc](https://lxml.de/2.2/tutorial.html)

Reply:
Thanks! No more loop needed.